openssl/src/pkey.rs

```diff
@@ -246,6 +250,58 @@ where
     {
         unsafe { ffi::EVP_PKEY_cmp(self.as_ptr(), other.as_ptr()) == 1 }
     }
+
+    /// Retrieves the length of the raw public key.
+    /// Please note that retrieving the raw public keys is only supported on keys of type
+    /// Ed25519, Ed448, X25519, X448.
+    /// This function will return 32 for Ed25519 and X25519 keys and 57 for Ed448 and X448 keys.
+    #[cfg(ossl111)]
+    pub fn get_raw_public_key_len(&self) -> Result<usize, ErrorStack> {
+        match self.id() {
+            Id::ED25519 | Id::X25519 => return Ok(32),
```

Comment:
Why are these hardcoded?

Reply:
RFC 8032 requires raw private/public keys for (X|Ed)(25519|448) to have the specified lengths. So there is no point in querying OpenSSL. It just saves a jump into the library and an error check. But I don't mind skipping it.

Reply:
Querying OpenSSL is literally the purpose of this library.